server/src/test/java/org/apache/iotdb/db/tools/TsFileSinglePathReadTest.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.iotdb.tsfile.file.metadata.enums.TSEncoding;
+import org.apache.iotdb.tsfile.read.TsFileSequenceReader;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.read.reader.chunk.ChunkReader;
+import org.apache.iotdb.tsfile.write.TsFileWriter;
+import org.apache.iotdb.tsfile.write.record.TSRecord;
+import org.apache.iotdb.tsfile.write.record.datapoint.FloatDataPoint;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TsFileSinglePathReadTest {
+
+  String fileName = "target/tsfileWriter-" + System.nanoTime();
+  TsFileWriter writer = null;
+  String path = "d1.s1";
+
+  @Before
+  public void setUp() {
+    try {
+      writer = new TsFileWriter(new File(fileName));
+      writeRecordAndClose();
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail(e.getMessage());
+    }
+  }
+
+  @After
+  public void tearDown() {
+    try {
+      Files.deleteIfExists(new File(fileName).toPath());
+    } catch (IOException e) {
+      e.printStackTrace();
+      fail(e.getMessage());
+    }
+  }
+
+  private void writeRecordAndClose() throws IOException, WriteProcessException {
+    addMeasurement();
+    writeTSRecord();
+    writer.close();
+  }
+
+  private void addMeasurement() {
+    try {
+      //String measurementId, TSDataType type, TSEncoding encoding,
+      //      CompressionType compressionType
+      writer.registerTimeseries(new Path("d1", "s1"),
+          new MeasurementSchema("s1", TSDataType.FLOAT, TSEncoding.RLE, CompressionType.SNAPPY));
+    } catch (WriteProcessException e) {
+      e.printStackTrace();
+      fail(e.getMessage());
+    }
+  }
+
+  private void writeTSRecord() throws IOException, WriteProcessException {
+    TSRecord record = new TSRecord(10000, "d1");
+    record.addTuple(new FloatDataPoint("s1", 5.0f));
+    writer.write(record);
+
+    record = new TSRecord(10001, "d1");
+    record.addTuple(new FloatDataPoint("s1", 5.0f));
+    try {
+      writer.write(record);
+    } catch (WriteProcessException e) {
+      assertTrue(e instanceof NoMeasurementException);
+    }
+  }
+
+  @Test
+  public void SinglePathReadTest() throws IOException {
+    int numCnt = 0;
+    try (TsFileSequenceReader reader = new TsFileSequenceReader(fileName)) {
+      // get the chunkMetaList of the specific path
+      List<ChunkMetadata> chunkMetadataList = reader
+          .getChunkMetadataList(new Path(path, true), true);
+      for (ChunkMetadata metadata : chunkMetadataList) {
+        ChunkReader chunkReader = new ChunkReader(reader.readMemChunk(metadata), null);
+        ChunkDataIterator chunkDataIterator = new ChunkDataIterator(chunkReader);
+        while (chunkDataIterator.hasNextTimeValuePair()) {
+          chunkDataIterator.nextTimeValuePair();
+          numCnt ++;
+        }
+      }
+    }
+    assertEquals(numCnt, 2);
```

Comment:
I think you should not write the same code again, you can extract the code in TsFileSinglePathRead as a public method, and in this UT, you can just test that method. Or if the class has been modified, this test may be forgotten to be changed.

Reply:
done